cpu/fe310/include/context_frame.h

```diff
@@ -0,0 +1,129 @@
+
+/*
+ * Copyright (C) 2017 JP Bonn
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    cpu_fe310
+ * @ingroup     cpu
+ * @brief       Freedom E cpu
+ * @{
+ *
+ * @file
+ * @brief       Thread context frame stored on stack.
+ *
+ * @author      JP Bonn
+ */
+
+#ifndef CONTEXT_FRAME_H
+#define CONTEXT_FRAME_H
+
+#if !defined(__ASSEMBLER__)
+#include <stdint.h>
+#include <assert.h>
+#endif /* __ASSEMBLER__ */
+
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   context_switch_frame stores the registers and PC for a context switch.
+ *
+ * This also defines context_switch_frame offsets for assembly language.  The
+ * structure is sized to maintain 16 byte stack alignment per the ABI.
+ * https://github.com/riscv/riscv-elf-psabi-doc
+ *
+ */
+#if !defined(__ASSEMBLER__)
+
+/* N.B.: update the definitions below if this changes */
+struct context_switch_frame {
```

Comment:
Add documentation for the struct. You don't need to document each member variable.